### DISCUSSION TITLE
The AI currently has two issues. It will go outside the submarine to search for a Deconstructor, regardless of the danger.

### DISCUSSION THREAD

**Discussion by KougamiYuriko** [FakeFishGames/Barotrauma#14855] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The first issue: If the structural scanner is in operation and the AI has something to decompose, it will try to find another Deconstructor. While the intention is good and it’s not a problem at the outpost, it can be frustrating since the AI won’t bring the decomposed items back to the ship. Surprisingly, it also does this during battles; the AI will attempt to find Deconstructors on sunken ships, suddenly donning a diving suit and swimming 900 meters alone to the wreck to decompose an empty canister, even if the Deconstructor inside the wreck isn’t operational. I was just stunned watching the AI swim off like that.

The second issue: After maintenance in the ballast tank, the AI won’t come out immediately; it prefers to swim back and forth in the ballast tank until it exhausts the oxygen tank.

### Reproduction steps

When the Deconstructor is in operation and multiple AIs need to decompose items, they will try to find a new Deconstructor. 
The AI also really enjoys standing still or swimming around in the ballast tank.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
[Finally addressed in our private repo](https://github.com/FakeFishGames/Barotrauma-development/commit/6b0616b0d9b9ffbbd1120a1db5ff1f4feb3aae80), fix coming up in the next unstable build. Sorry for it taking so ridiculously long for us to look into this!

**Reply by KougamiYuriko** [reply]
Thank you for your response and for resolving the issue.
